test/lock.test.ts

```diff
@@ -44,4 +48,52 @@ describe("Lock", () => {
     expect(sharedResource).toContain("b");
     expect(sharedResource).toEqual(r);
   });
+
+  test("Lock store behaves correctly when acquiring and releasing locks", async () => {
+    const acquireResult = await lockStore.tryAcquire("resource-id-1", "execution-id-1");
+    expect(acquireResult).toBe(true);
+
+    if (useDurable) {
+      // should get error 403
+      await expect(lockStore.tryAcquire("resource-id-1", "execution-id-2")).rejects.toThrow();
+    } else {
+      const secondAcquireResult = await lockStore.tryAcquire("resource-id-1", "execution-id-2");
+      expect(secondAcquireResult).toBe(false);
+    }
+
+    await lockStore.release("resource-id-1", "execution-id-1");
+
+    if (useDurable) {
+      // should result in an error 404
+      await expect(lockStore.release("resource-id-1", "execution-id-1")).rejects.toThrow();
+    } else {
+      lockStore.release("resource-id-2", "execution-id-1");
+    }
+  });
+
+  test("Lock expiration works as expected", async () => {
+    if (!useDurable) {
+      return;
+    }
+
+    // Acquire a lock with a short expiration time
+    const acquireResult = await lockStore.tryAcquire("resource-id-3", "execution-id-1");
+    expect(acquireResult).toBe(true);
+
+    // Wait for the lock to expire
+    await new Promise((resolve) => setTimeout(resolve, 70000)); // Assuming lockExpiry is 60000
```

Comment:
We should have a way to set the initial expiry to 0 so that we don't have to wait a minute for the tests to run